cpp/ql/src/experimental/Security/CWE/CWE-193/InvalidPointerDeref.ql

```diff
@@ -160,13 +241,40 @@ pragma[nomagic]
 predicate pointerAddInstructionHasBounds(
   PointerAddInstruction pai, DataFlow::Node sink1, DataFlow::Node sink2, int delta
 ) {
-  exists(Instruction right |
+  InterestingPointerAddInstruction::isInteresting(pragma[only_bind_into](pai)) and
+  exists(Instruction right, Instruction instr2 |
     pai.getRight() = right and
     pai.getLeft() = sink1.asInstruction() and
-    bounded1(right, sink2.asInstruction(), delta)
+    instr2 = sink2.asInstruction() and
+    bounded1(right, instr2, delta) and
+    not right = Barrier2::getABarrierInstruction(delta) and
+    not instr2 = Barrier2::getABarrierInstruction(delta)
   )
 }
 
+module InterestingPointerAddInstruction {
```

Comment:
I assume you saw some performance problems that made this necessary?

Reply:
Yeah, exactly. Remember how `pointerAddInstructionHasBounds` was super slow because the predicate was gigantic on Wireshark because there was sooo many bounds available for some instructions? It became even slower on this PR (see [this earlier DCA run](https://github.com/github/codeql-dca-main/tree/data/MathiasVP/pr-13559-b87bf4__nightly__memory-corrupti/reports#summaries-1)), but after this commit it's [a lot faster than before](https://github.com/github/codeql-dca-main/blob/data/MathiasVP/pr-13559-ef383a__nightly__memory-corrupti/reports/readme.md#summaries-1).